docker/bin/zookeeperStart.sh

```diff
@@ -138,13 +138,18 @@ fi
 ZOOCFGDIR=/data/conf
 export ZOOCFGDIR
 if [[ ! -d "$ZOOCFGDIR" ]]; then
-  echo Copying /conf contents to writable directory, to support Zookeeper dynamic reconfiguration
   mkdir $ZOOCFGDIR
-  cp -f /conf/zoo.cfg $ZOOCFGDIR
-  cp -f /conf/log4j.properties $ZOOCFGDIR
-  cp -f /conf/log4j-quiet.properties $ZOOCFGDIR
-  cp -f /conf/env.sh $ZOOCFGDIR
 fi
+echo Copying /conf contents to writable directory, to support Zookeeper dynamic reconfiguration
+cp -f /conf/log4j.properties $ZOOCFGDIR
+cp -f /conf/log4j-quiet.properties $ZOOCFGDIR
+cp -f /conf/env.sh $ZOOCFGDIR
+# create or refresh $ZOOCFGDIR/zoo.cfg appending to /conf/zoo.cfg
+# the last dynamicConfigFile entry (if available)
+cat \
+<<<$(cat /conf/zoo.cfg) \
+<<<$(grep 'dynamicConfigFile=' $ZOOCFGDIR/zoo.cfg 2>/dev/null) \
```

Comment:
If we already have the `dynamicConfigFile` property (which will happen in most of the cases), wouldn't this result in a file with both the new and the old `dynamicConfigFile` property? Because we are writing the full content of `/conf/zoo.cfg` + another `dynamicConfigFile`. I think we should use `sed` or something similar to replace the old `dynamicConfigFile`

Reply:
updated to cover for this